solidity/contracts/bridge/Wallets.sol

```diff
@@ -624,4 +624,34 @@ library Wallets {
 
         closeWallet(self, walletPubKeyHash);
     }
+
+    /// @notice Notifies about a timed out moving funds process. Terminates
+    ///         the wallet in result.
+    /// @param walletPubKeyHash 20-byte public key hash of the wallet
+    /// @dev Requirements:
+    ///      - The wallet must be in the MovingFunds state
+    ///      - The moving funds timeout must be actually exceeded
+    function notifyWalletMovingFundsTimeout(
+        BridgeState.Storage storage self,
+        bytes20 walletPubKeyHash
+    ) internal {
+        Wallet storage wallet = self.registeredWallets[walletPubKeyHash];
+
+        require(
+            wallet.state == WalletState.MovingFunds,
+            "ECDSA wallet must be in MovingFunds state"
+        );
+
+        require(
+            /* solhint-disable-next-line not-rely-on-time */
+            block.timestamp >
+                wallet.movingFundsRequestedAt + self.movingFundsTimeout,
+            "Moving funds has not timed out"
+        );
```

Comment:
Based on [this conversation](https://github.com/keep-network/tbtc-v2/pull/208/files#discussion_r854996501), I think this part should be in `MovingFunds` library. I think we also don't need `notifyWalletMovingFundsTimeout` in `Wallets` at all. It is fine if we call `terminateWallet` directly. IMO, slashing in `MovingFunds` is fine as well.